app/controllers/alchemy/admin/pages_controller.rb

```diff
@@ -171,9 +171,10 @@ def unlock
 
       def visit
         @page.unlock!
-        redirect_to show_page_path(
+        redirect_to show_page_url(
           urlname: @page.urlname,
-          locale: prefix_locale? ? @page.language_code : nil
+          locale: prefix_locale? ? @page.language_code : nil,
+          host: @page.site.host == "*" ? request.host : @page.site.host
```

Comment:
Unfortunately this won't work if:

1. You visit your local site via mysite.dev
2. Change the site via "Site select" to a default Rails dev site (`http://localhost:3000`)
3. Visit the page via "Visit page" button 

=> 404

Because the resulting url is `http://localhost/home` instead of `http://localhost:3000/home`

/c @wdspkr